lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -53,27 +53,35 @@ def _get(self, params: ApiParams, _) -> Response:
 
         GET /api/v1/datasets/inventory/{dataset}/{path}
         """
+        try:
+            path = params.uri["path"]
+        except KeyError:
+            path = None
 
         dataset = params.uri["dataset"]
-        path = params.uri["path"]
-
         file_tree = FileTree(self.config, self.logger)
         try:
             tarball = file_tree.find_dataset(dataset.resource_id)
         except TarballNotFound as e:
             raise APIAbort(HTTPStatus.NOT_FOUND, str(e))
-        dataset_location = tarball.unpacked
-        if dataset_location is None:
-            raise APIAbort(HTTPStatus.NOT_FOUND, "The dataset is not unpacked")
-        file_path = dataset_location / path
-        if file_path.is_file():
-            return send_file(file_path)
-        elif file_path.exists():
-            raise APIAbort(
-                HTTPStatus.UNSUPPORTED_MEDIA_TYPE,
-                "The specified path does not refer to a regular file",
-            )
+
+        if path is None:
+            tarball_path = tarball.tarball_path
+            return send_file(tarball_path)
+
         else:
-            raise APIAbort(
-                HTTPStatus.NOT_FOUND, "The specified path does not refer to a file"
-            )
+            dataset_location = tarball.unpacked
+            if dataset_location is None:
+                raise APIAbort(HTTPStatus.NOT_FOUND, "The dataset is not unpacked")
+            file_path = dataset_location / path
+            if file_path.is_file():
+                return send_file(file_path)
+            elif file_path.exists():
+                raise APIAbort(
+                    HTTPStatus.UNSUPPORTED_MEDIA_TYPE,
+                    "The specified path does not refer to a regular file",
+                )
+            else:
+                raise APIAbort(
+                    HTTPStatus.NOT_FOUND, "The specified path does not refer to a file"
+                )
```

Comment:
We assume that these checks probably aren't relevant for the tarball itself, but it wouldn't hurt to make them. That would also cut down on the amount of conditional code:

```
        if path is None:
            file_path = tarball.tarball_path
        else:
            dataset_location = tarball.unpacked
            if dataset_location is None:
                raise APIAbort(HTTPStatus.NOT_FOUND, "The dataset is not unpacked")
            file_path = dataset_location / path

        if file_path.is_file():
            return send_file(file_path)
        [...]
```

This also reduces to a single return point, which can make the code a lot easier to read.